### ISSUE TITLE
Ensures in override not proven

### ISSUE THREAD

**Issue by Nathan-022** [Microsoft/CodeContracts#265]
I have overridden TypeConverter.ConvertFrom and attempted to Ensure the return value is not null. The base TypeConverter class doesn't ensure this, so I am making the postcondition stronger in my custom implementation.

From my understanding, inheritance works like:
* If a subclass overrides the behavior of a method of a base class (or implements and interface), the preconditions it specifies can only be equivalent to or weaker than the base class
* If a subclass overrides the behavior of a method of a base class (or implements and interface), the postconditions it specifies can only be equivalent to or stronger than the base class.

From this description, I should be able to have a stronger postcondition in the subclass (non null return) and have it respected by the analysis tool, however in practice this isn't working.

Is my interpretation incorrect or is this a bug in the static analysis?

Sample code:

```csharp
CustomTypeConverter converter = new CustomTypeConverter();
object converted = converter.ConvertFrom(null, CultureInfo.CurrentCulture, value);
Contract.Assume(converted != null, "Contracts analysis doesn't realise that the non null return was ensured, the below code gives an 'unboxing null reference' warning");
this = (CustomType)converted;
```

Inside the typeconverter, something like:

```csharp
public override object ConvertFrom(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value)
{
	Contract.Ensures(Contract.Result<object>() != null);
	return "example";
}
```

**Comment by mike-barnett** [comment]
You are correct in the theory of modifying contracts. Note that just as an implementation decision, Code Contracts does not allow you to weaken preconditions, but only to strengthen postconditions.

It looks to me as if that is a bug in the static checker. However, now that Francesco is not working on the project any longer, it isn't clear if there is anyone who will be able to fix it.
